dozent/downloader_tools.py

```diff
@@ -1,100 +1,90 @@
-import multiprocessing
-import os
 import time
-import random
+import sys
+from hurry.filesize import size
 from typing import Tuple
+from threading import Lock
 from pySmartDL import SmartDL
 
+# Used for tracking and displaying download progress bars
+global global_progress_tracker
+global_progress_tracker = [(0, 0, "0", 0, "0")]
+global global_download_size
+global_download_size = 0
+
+global global_final_download_size
+global_final_download_size = 0
+
+
 class DownloaderTools:
     __instance__ = None
 
     def __init__(self):
         if DownloaderTools.__instance__ is None:
             DownloaderTools.__instance__ = self
-
         else:
-            raise RuntimeError(f"Singleton {self.__class__.__name__} class is created more than once!")
+            raise RuntimeError(
+                f"Singleton {self.__class__.__name__} class is created more than once!"
+            )
 
     @staticmethod
-    def _make_progress_status(downloader_obj: SmartDL) -> Tuple[float, str, str]:
-        """Function to make progress bar
-        :param downloader_obj: SmartDL object that's currently downloading a file
-        :return: the progress percentage in [0,100] and a string prefix/suffix to be output before/after a progress bar.
-        """
-
-        status = downloader_obj.get_status()
-        filesize = downloader_obj.get_final_filesize() >> 20
-
-        try:
-            dl_size = downloader_obj.get_dl_size() >> 20
-            speed = downloader_obj.get_speed(human=True)
-            eta = downloader_obj.get_eta(human=True)
-            progress_percentage = 100.0 * downloader_obj.get_progress()
+    def _get_individual_download_stats(
```

Comment:
Don't forget to add comments/docs to your code

Reply:
On it